python/lsst/analysis/tools/tasks/limitingSurfaceBrightnessAnalysis.py

```diff
@@ -0,0 +1,295 @@
+
+    Reference: Roman, J., Trujillo, I., & Montes, M., 2020, A & A, 644, 42
+    """
+
+    ConfigClass = LimitingSurfaceBrightnessConfig
+    _DefaultName = "limitingSurfaceBrightness"
+
+    def __init__(self, initInputs=None, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def runQuantum(self, butlerQC, inputRefs, outputRefs):
+        inputs = butlerQC.get(inputRefs)
+        limiting_surface_brightness_struct = self.run(**{k: v for k, v in inputs.items() if k != "skymap"})
+        butlerQC.put(limiting_surface_brightness_struct, outputRefs)
+
+    def run(self, data, photoCalib, wcs):
+        """Output a table of per-detector or per-patch limiting surface
+        brightnesses for the input visit or tract coadd
+
+        Parameters
+        ----------
+        data : `list`
+            List of dicts with information respecting the extracted source
+            detection catalogues
+        photoCalib : `list`
+            List of dicts with information respecting the extracted image
+            photometric calibrations
+        wcs : `list`
+            List of dicts with information respecting the extracted image
+            world coordinate systems
+
+        Returns
+        -------
+        `pipe.base.Struct` containing `astropy.table.Table`
+            Table containing per-detector or per-patch limiting surface
+            brightness values
+        """
+        # Generate lookup tables
+        lookup_photoCalib = {x.dataId: x for x in photoCalib}
+        lookup_wcs = {x.dataId: x for x in wcs}
+
+        # Retrieve the source catalogue and photometric band if required
+        source_catalogue = data[0].get()
+        if "band" in self.config.inputCalibDimensions:
+            band = photoCalib[0].dataId["band"] + "_"
+        else:
+            band = ""
+
+        # Set up the global table
+        limiting_surface_brightness_table = Table(
+            names=["imageID", "limiting_surface_brightness"],
+            dtype=[np.int64, np.float64],
+        )
+
+        # Loop over the detector or patch calibration data
+        for dataId in lookup_photoCalib.keys():
+            # And the photometric calibration
+            instFluxToMagnitude = lookup_photoCalib[dataId].get().instFluxToMagnitude(1)
+            # And the pixel scale
+            pxScale = lookup_wcs[dataId].get().getPixelScale().asArcseconds()
+
+            # Isolate the sky sources per image, 9px radius apertures
+            idKey = "detector" if "detector" in dataId else "patch"
+            skyKey = "sky_source" if "detector" in dataId else "sky_object"
+            isImage = source_catalogue[idKey] == dataId[idKey]
+            isSky = source_catalogue[skyKey] > 0
+            skySources = source_catalogue[isImage & isSky][band + "ap%02dFlux" % (self.config.apertureSize)]
+
+            # Derive the clipped standard deviation of sky sources in nJy
+            nPix = np.pi * 9**2  # Number of pixels within the circular aperture
```

Comment:
This version has the 9 here but I am not sure which version of this code is the correct one.

Reply:
Replaced this with the config value.